src/helpers/in-memory-cache.ts

```diff
@@ -0,0 +1,48 @@
+import type { VirtualCurrencies } from "../entities/virtual-currencies";
+
+interface CacheEntry<T> {
+  data: T;
+  timestamp: number;
+}
+
+export class InMemoryCache {
```

Comment:
I think this is fine since we don't use localStorage but pinging @nicfix to confirm. We've been trying to avoid caching most things in web until now.